### PR TITLE
Update sonoff_basic.rst

### PR DESCRIPTION
**Pull request by workingmanrob**
Use same recommended device config section as the info here - https://esphome.io/devices/sonoff.html

## Description: 

Slight difference/conflict as to what to set as the sonoff basic board type.  Just converted all my sonoff basics to use the same esphome block as the recommendation for the generic sonoff and everything is working well.

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: devices/sonoff_basic.rst [hunk 1]
```diff
@@ -54,7 +54,8 @@ exposes all of the basic functions.
     esphome:
       name: <NAME_OF_NODE>
       platform: ESP8266
-      board: esp01_1m
+      board: esp8285
+      arduino_version: 2.4.2
```
Comment (frenck) [inline devices/sonoff_basic.rst:58]: I wonder if this Arduino downgrade is still needed. This is based on information from 2019...

Comment (workingmanrob) [reply devices/sonoff_basic.rst:58]: I have a serial console connected to the one on my desk.

Will test without that arduino version line this afternoon and let you know how it goes..

Comment (workingmanrob) [reply devices/sonoff_basic.rst:58]: Did not go well.

Never reconnected to wifi and lucky for me I had the serial console so I could easily flash back the 2.4.2 version.

These devices are very finicky but using that declared arduino_version does seem to make a difference for the better.

Comment (frenck) [reply devices/sonoff_basic.rst:58]: Thanks! 👍

**Review by frenck — APPROVED** [review]
Thanks, @workingmanrob 👍

**Pull request merged**